testflinger_agent/client.py

```diff
@@ -55,6 +58,28 @@ def _requests_retry(self, retries=3):
         session.mount("https://", adapter)
         return session
 
+    def _configure_influx(self):
+        """Configure InfluxDB client using environment variables.
+
+        :return: InfluxDBClient object
+        """
+        # required influxdb env vars
+        req_env_vars = ["INFLUX_HOST", "INFLUX_USER", "INFLUX_PW"]
```

Comment:
So, it's actually plausible that you might have an INFLUX_HOST, but the USER and PW could be empty strings. It just so happens that in our case, we do actually auth for this, but in some cases it might be empty - for instance, a test setup, or inside a k8s namespace. Let's support that possibility by not requiring the USER/PW, but we must also detect if something went wrong when authenticating, and log an error if that happens.